src/models/vulnerability/vulnerability.ts

```diff
@@ -97,7 +99,23 @@ export class Vulnerability {
   get bomRef (): BomRef {
     return this.#bomRef
   }
+
+  compare (other: Vulnerability): number {
+    const bomRefCompare = this.bomRef.compare(other.bomRef)
+    if (bomRefCompare !== 0) {
+      return bomRefCompare
+    }
+    if (this.source !== undefined && other.source !== undefined) {
+      return this.source.compare(other.source)
+    }
+
+    /* eslint-disable @typescript-eslint/strict-boolean-expressions -- run compares in weighted order */
+    return (this.id ?? '').localeCompare(other.id ?? '') ||
+      (this.description ?? '').localeCompare(other.description ?? '') ||
+      (this.detail ?? '').localeCompare(other.detail ?? '')
```

Comment:
Not sure if I need to add checks for all the properties of the `Vulnerability` model here.

Reply:
me neither. just do what feels right at this point.
if somebody feels the need to improve, they can request changes later.